language/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -493,55 +630,154 @@ impl DependencyGraph {
         package_path: &Path,
         dependency_cache: &mut DependencyCache,
         external_requests: &mut Vec<ExternalRequest>,
+        overrides: &mut BTreeMap<PM::PackageName, Package>,
         progress_output: &mut Progress,
     ) -> Result<()> {
         let from = package.package.name;
-        for (to, dep) in &package.dependencies {
-            match dep {
-                PM::Dependency::External(resolver) => external_requests.push(ExternalRequest {
-                    mode: DependencyMode::Always,
-                    from,
-                    to: *to,
-                    resolver: *resolver,
-                    pkg_path: package_path.to_path_buf(),
-                }),
-                PM::Dependency::Internal(dep) => self.resolve_internally(
-                    DependencyMode::Always,
-                    from,
-                    *to,
-                    parent,
-                    dep.clone(),
-                    dependency_cache,
-                    external_requests,
-                    progress_output,
-                )?,
+
+        self.extend_with_dependencies(
+            DependencyMode::Always,
+            &package.dependencies,
+            from,
+            parent,
+            package_path,
+            dependency_cache,
+            external_requests,
+            overrides,
+            progress_output,
+        )?;
+
+        self.extend_with_dependencies(
+            DependencyMode::DevOnly,
+            &package.dev_dependencies,
+            from,
+            parent,
+            package_path,
+            dependency_cache,
+            external_requests,
+            overrides,
+            progress_output,
+        )?;
+
+        Ok(())
+    }
+
+    fn extend_with_dependencies<Progress: Write>(
+        &mut self,
+        mode: DependencyMode,
+        dependencies: &PM::Dependencies,
+        from: Symbol,
+        parent: &PM::DependencyKind,
+        package_path: &Path,
+        dependency_cache: &mut DependencyCache,
+        external_requests: &mut Vec<ExternalRequest>,
+        overrides: &mut BTreeMap<PM::PackageName, Package>,
+        progress_output: &mut Progress,
+    ) -> Result<()> {
+        // partition dep into overrides and not
+        let (overridden_deps, deps): (Vec<_>, Vec<_>) =
+            dependencies.iter().partition(|(_, dep)| {
+                if let PM::Dependency::Internal(d) = dep {
+                    d.dep_override
+                } else {
+                    false
+                }
```

Comment:
Not 100% sure this is actually neater, but:

```suggestion
                matches!(dep, PM::Dependency::Internal(InternalDependency {
                    dep_override: true,
                    ..
                }))
```